editor/sidebar/post-taxonomies/index.js

```diff
@@ -24,58 +24,37 @@ import { toggleSidebarPanel } from '../../actions';
  */
 const PANEL_NAME = 'post-taxonomies';
 
-class PostTaxonomies extends Component {
-	constructor() {
-		super( ...arguments );
+function PostTaxonomies( { postType, taxonomies, isOpened, onTogglePanel } ) {
+	const availableTaxonomies = !! taxonomies.data
+		? Object.values( taxonomies.data ).filter( ( taxonomy ) => taxonomy.types.indexOf( postType ) !== -1 )
+		: [];
 
-		this.state = {
-			taxonomies: [],
-		};
-	}
-
-	componentDidMount() {
-		this.fetchTaxonomies = new wp.api.collections.Taxonomies()
-			.fetch()
-			.done( ( taxonomies ) => {
-				this.setState( { taxonomies: Object.values( taxonomies ) } );
-			} );
-	}
-
-	componentWillUnmout() {
-		this.fetchTaxonomies.abort();
+	if ( ! availableTaxonomies.length ) {
```

Comment:
I think we should seek to avoid content flickering in and out of the sidebar as the page loads. Especially if we're more likely to have taxonomies than not, we could at least show the panel heading, even if the body can't be known yet. Alternatively, #2501 was designed for this purpose, and we could consider preloading taxonomies.